### PR TITLE
Do not create if device node does not exist

### PR DESCRIPTION
**Pull request by SmithChart**
This PR fixes a bug in which a file is created in ``/dev/`` instead of a device node when the ``usbsdmux``-tool is called as root with a non-existing ``/dev/sg*`` parameter.

And while I was on it i also improved the error handling in the CLI. With this change the CLI only print the error message if a well-known Exception occurs.

### CODE REVIEW

File: usbsdmux/__main__.py [hunk 1]
```diff
@@ -64,17 +64,38 @@ def main():
...
+
+        elif mode == "host":
+            ctl.mode_host()
+
+        elif mode == "get":
+            print(ctl.get_mode())
+
+    except FileNotFoundError as fnfe:
+        print(fnfe, file=sys.stderr)
+        sys.exit(fnfe.errno)
```
Comment (jluebbe) [inline usbsdmux/__main__.py:82]: errno is unusual as an exit code. If you don't have anything else defined, just use sys.exit(1).

File: usbsdmux/__main__.py [hunk 2]
```diff
@@ -64,17 +64,38 @@ def main():
...
+            print(
+                f"Does '{args.sg}' really point to an USB-SD-Mux?",
+                file=sys.stderr,
+            )
+            sys.exit(ose.errno)
+        else:
+            raise ose
+
+    except Exception as exe:
+        raise exe
```
Comment (jluebbe) [inline usbsdmux/__main__.py:98]: Why is this needed? It shouldn't do anything...

File: usbsdmux/__main__.py [hunk 3]
```diff
@@ -64,17 +64,38 @@ def main():
...
+            print(ctl.get_mode())
+
+    except FileNotFoundError as fnfe:
+        print(fnfe, file=sys.stderr)
+        sys.exit(fnfe.errno)
+    except PermissionError as perr:
+        print(perr, file=sys.stderr)
+        sys.exit(perr.errno)
+    except OSError as ose:
+        if ose.errno == 25:
```
Comment (hnez) [inline usbsdmux/__main__.py:87]: I suggest not using the magic number here but instead:

    import errno

    if ose.errno == errno.ENOTTY:
        # An error occured when calling ioctl()
        print(ose, file=sys.stderr)
        …

**Review by hnez — DISMISSED** [review]
I did not test this with an actual USB-SD-Mux, but as this is exactly the corner-case this MR wants to cover that should be fine.

I agree with @jluebbe's remarks and added one about the magic errno number in the source.
If these comments are addressed I approve.

File: usbsdmux/__main__.py (hunk 1 above)
Comment (SmithChart) [reply usbsdmux/__main__.py:82]: It is fine to just return sys.exit(1). I will change it this way.

File: usbsdmux/__main__.py (hunk 2 above)
Comment (SmithChart) [reply usbsdmux/__main__.py:98]: You are right. No need to raise it manually.

File: usbsdmux/__main__.py (hunk 3 above)
Comment (SmithChart) [reply usbsdmux/__main__.py]: Totally right :) Thx

**Review by hnez — APPROVED** [review]
Looks good to me
